src/goto-programs/loop_unroll.cpp

```diff
@@ -0,0 +1,171 @@
+#include <goto-programs/loop_unroll.h>
+
+bool unsound_loop_unroller::runOnLoop(loopst &loop, goto_programt &goto_program)
+{
+  int bound = get_loop_bounds(loop);
+  if(bound <= 0)
+    return false; // Can't unroll this
+
+  // Get loop exit goto number
+  goto_programt::targett loop_exit = loop.get_original_loop_exit();
+  loop_exit->make_skip();
+
+  /* TODO: If we want a sound version: loop_exit++ so we have the skip instruction
```

Comment:
Do we need to keep these comments?

Reply:
Yeah, this comment sounds misleading, does it mean that our algorithm is unsound?

Reply:
> does it mean that our algorithm is unsound?

There are two classes: 
 - unsound_loop_unroller: which is unsound.
 - bounded_loop_unroller: which inherits from unsound_loop_unroller but it is sound.

Now I see that it looks like everything is unsound. Any suggestions? Maybe renaming it to loop_unroller will do the trick.

The unroll removes the check for the loop exit, it is unsound if applied directly. This means that the goto instruction that will check for the loop condition e.g. "IF i < 10 goto 2" will be removed. 

However, in ESBMC this is not used directly, it is used only if the loop has known bounds, which makes it sound.

I've created this way to make it easier to explore unroll techniques for unbounded loops as well, since the same principle still applies.

Reply:
I' ve replaced the comment